test/testcanvas.go

```diff
@@ -43,90 +44,109 @@ type testCanvas struct {
 	propertyLock sync.RWMutex
 }
 
-func (c *testCanvas) Content() fyne.CanvasObject {
-	c.propertyLock.RLock()
-	defer c.propertyLock.RUnlock()
+// Canvas returns a reusable in-memory canvas used for testing
+func Canvas() fyne.Canvas {
+	if dummyCanvas == nil {
+		dummyCanvas = NewCanvas()
+	}
 
-	return c.content
+	return dummyCanvas
 }
 
-func (c *testCanvas) SetContent(content fyne.CanvasObject) {
-	c.propertyLock.Lock()
-	c.content = content
-	c.propertyLock.Unlock()
-
-	if content == nil {
-		return
+// LaidOutObjects returns all fyne.CanvasObject starting at the given fyne.CanvasObject which is laid out previously.
+func LaidOutObjects(o fyne.CanvasObject) (objects []fyne.CanvasObject) {
```

Comment:
Seems like a strange place to move this function to

Reply:
I thought we agreed to exported package functions before any package type functions, not?

Reply:
Pretty sure it’s type -> constructor -> exported type func -> unexported type func.
We didn’t put exported package funcs in the list I don’t think, I would imagine that it’s either at the top or the bottom of the file, but not in between type declarations and type functions.

Reply:
Okay, so I propose the top of the file :).